src/Microsoft.OData.Core/JsonLight/ODataJsonLightReader.cs

```diff
@@ -2256,6 +2259,109 @@ private void EndEntry()
             }
         }
 
+        /// <summary>
+        /// For response de-serialization, restore the null values for properties that are omitted
+        /// by the omit-values=nulls preference header.
+        /// </summary>
+        private void RestoreOmittedNullValues()
+        {
+            // Restore omitted null value only when processing response and when
+            // Preference header omit-values=nulls is specified.
+            if (this.jsonLightInputContext.ReadingResponse
+                && this.jsonLightInputContext.MessageReaderSettings.NullValuesOmitted)

```

Comment:
should this be encapsulated in a method on the input context, similar to the ShouldWriteNullValues() method on the output context?

Reply:
Moved to ODataReaderCore.ShouldRestoreNullValues().

---
In reply to: [186532720](https://github.com/OData/odata.net/pull/1106#discussion_r186532720) [](ancestors = 186532720)